solidity/migrations/2_sausage_migration.js

```diff
@@ -1,20 +1,143 @@
 const SausageTokens = artifacts.require('./SausageTokens.sol')
 const Auction = artifacts.require('./Auction.sol')
 
-module.exports = async function(deployer) {
-  await deployer.deploy(SausageTokens, 'SausageTokens', 'Bratwurst')
-  const sausage = await SausageTokens.deployed()
-  await deployer.deploy(
-    Auction,
-    '0xe207259495d9de39e92e8b280a85b4d845906ce6',
-    1234,
-    'Bratwurst'
-  )
-  const auction = await Auction.deployed()
-}
+const Web3 = require('web3')
+
+module.exports = async function(deployer, network) {
+  let web3
+  let sausageID = 1234
+  let sausageSymbol = 'Bratwurst'
+  let sausageDescription = 'Bratwurst - Rare'
+  let sausageInstance
+  let managerAccount
+  let accounts
+
+  if (network === 'development') {
+    web3 = new Web3(new Web3.providers.HttpProvider('http://localhost:8545'))
+    managerAccount = web3.accounts[0]
+  } else if (network === 'kovan') {
+    // do something similar
+  } else if (network === 'parity') {
+    web3 = new Web3(new Web3.providers.HttpProvider('http://localhost:8545'))
+    await web3.eth.getAccounts().then(acc => {
+      accounts = acc
+      managerAccount = acc[1]
+    })
+  }
+
+  return deployer
+    .deploy(SausageTokens, 'SausageTokens', 'Bratwurst', {
+      from: managerAccount
+    })
+    .then(async () => {
+      sausageInstance = await SausageTokens.deployed()
+      let sausageAddress = sausageInstance.address
+      await deployer.deploy(
+        Auction,
+        managerAccount,
+        sausageID,
+        sausageSymbol,
+        sausageAddress,
+        { from: managerAccount }
+      )
+      let auctionInstance = await Auction.deployed()
+      let auctionAddress = auctionInstance.address
+      await sausageInstance.mintUniqueTokenTo(
+        auctionAddress,
+        sausageID,
+        sausageDescription
+      )
+      await auctionInstance.startAuction(1000, { from: managerAccount }) //auction has 1000 seconds
+
+      let originalBid = 1
```

Comment:
Everything below here we can actually put into a script , even though i put it in here for convenience, it probably doesn't belong here